### PR TITLE
Frontend/create season updates

### PR DESCRIPTION
**Pull request by Paalar**
- Lets admin update current season or create a new one
- Lets admin access a season through url or link with #271 
- Re-works  how application season works in the reducer
- Relies on backend from #256

### CODE REVIEW

File: frontend/src/components/CreateSeason/index.tsx [hunk 1]
```diff
@@ -168,11 +201,15 @@ class _CreateSeason extends Component<Props, IState> {
 }
 
 const mapStateToProps = (state: IStore) => ({
+  currentSeason: state.applicationSeason.currentSeason,
+  id: state.applicationSeason.id,
   submitted: state.applicationSeason.submitted,
```
Comment (PeterRydberg) [inline frontend/src/components/CreateSeason/index.tsx]: The fields above could be combined into applicationSeason and sent as one in props, but that might not be necessary

**Comment by PeterRydberg** [comment]
Fix merge conflict and type error (see CI log), then I'll approve

**Comment by Paalar** [comment]
> Fix merge conflict and type error (see CI log), then I'll approve

Fixed merge conflicts.
The error has previously been fixed, but the CI doesn't understand. You can run all the commans (npm run lint, npm run test, flake8, python3 -m pytest) and they will all pass.

File: frontend/src/components/CreateSeason/index.tsx (hunk 1 above)
Comment (Paalar) [reply frontend/src/components/CreateSeason/index.tsx]: Agreed, but I preferred splitting them, but can combine them if others find it better as well.

Comment (PeterRydberg) [reply frontend/src/components/CreateSeason/index.tsx]: I think it's fine, I'll approve

**Comment by Adrianht** [comment]
When having already created a season, nothing happens if you click create new season with different dates

**Comment by Adrianht** [comment]
When having already created a season, you can't update it, by setting new dates and clicking update season

**Comment by Paalar** [comment]
> After creating a season, you can't go back to create season in the sidebar

Fixed.

> When having already created a season, nothing happens if you click create new season with different dates

This is a problem with how we do seasons on main page. Not part of this pr.


> When having already created a season, you can't update it, by setting new dates and clicking update season

Fixed.

**Pull request closed without merging**